### PR TITLE
[example] Update mpm_lagrangian_forces.py to new syntax

### PR DESCRIPTION
**Pull request by yuanming-hu**
This PR does not change its behavior.

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

**Comment by archibate** [comment]
I found that the '1d cloth' seems to be 'stick' on the surface of ball in latest master on my end, is this intentional?

File: examples/mpm_lagrangian_forces.py [hunk 1]
```diff
@@ -90,23 +82,23 @@ def grid_op():
...
-            if j > n_grid - bound and grid_v(1)[i, j] > 0:
-                grid_v(1)[i, j] = 0
+            if i < bound and grid_v[i, j].x < 0:
+                grid_v[i, j].x = 0
+            if i > n_grid - bound and grid_v[i, j].x > 0:
+                grid_v[i, j].x = 0
+            if j < bound and grid_v[i, j].y < 0:
+                grid_v[i, j].y = 0
+            if j > n_grid - bound and grid_v[i, j].y > 0:
+                grid_v[i, j].y = 0
```
Comment (archibate) [inline examples/mpm_lagrangian_forces.py:101]: Cool! Will we adapt this usage in other examples, like mpm128?

File: examples/mpm_lagrangian_forces.py [hunk 2]
```diff
@@ -120,9 +112,10 @@ def g2p():
 
         for i in ti.static(range(3)):
             for j in ti.static(range(3)):
-                dpos = ti.cast(ti.Vector([i, j]), ti.f32) - fx
-                g_v = grid_v[base(0) + i, base(1) + j]
-                weight = w[i](0) * w[j](1)
+                I = ti.Vector([i, j])
+                dpos = ti.cast(I, ti.f32) - fx
```
Comment (archibate) [inline examples/mpm_lagrangian_forces.py:116]:
```suggestion
                dpos = float(I) - fx
```

File: examples/mpm_lagrangian_forces.py [hunk 3]
```diff
@@ -73,12 +65,12 @@ def p2g():
...
         for i in ti.static(range(3)):
             for j in ti.static(range(3)):
-                offset = ti.Vector([i, j])
-                dpos = (ti.cast(ti.Vector([i, j]), ti.f32) - fx) * dx
-                weight = w[i](0) * w[j](1)
-                grid_v[base + offset] += weight * (p_mass * v[p] - x.grad[p] +
-                                                   affine @ dpos)
-                grid_m[base + offset] += weight * p_mass
+                I = ti.Vector([i, j])
+                dpos = (ti.cast(I, ti.f32) - fx) * dx
```
Comment (archibate) [inline examples/mpm_lagrangian_forces.py:69]:
```suggestion
                dpos = (float(I) - fx) * dx
```

File: examples/mpm_lagrangian_forces.py (hunk 1 above)
Comment (yuanming-hu) [reply examples/mpm_lagrangian_forces.py:101]: Sounds good! But let me focus this PR on a single file.

**Pull request merged**